context-and-compliance/02-testing.md

```diff
@@ -0,0 +1,31 @@
+# 2 - Testing
+
+## 2.1 Test code coverage of 90% minimum
+
+#### Context & justification
+
+Ensuring **Test Driven Development (TDD)** is widely practiced across Made Tech, tests should be written for _every_ piece of code. Test code coverage of 90% minimum **should** indicate that 10% of code is not covered by tests. In some cases, tests cannot be written for every piece of code so 90% is chosen as the minimum. However, all project teams should aim to write tests for 100% of their code.
```

Comment:
There aren't really any cases where tests can't be written (though I defer to @craigjbass on this one).

The reason 90% was chosen historically was to accommodate legacy projects that were structured in such a way that did actually prevent fully testing things.


Might be worth making it so greenfield projects have a bar of 100%.

Reply:
I will mention the legacy projects as part of the "cases" and add that new projects should aim towards 100% of test code coverage. Thank you Anthony. :)

Reply:
Resolved in https://github.com/madetech/productionisation/pull/57/commits/42fcddd3aeca0c94678018ebf83550748d0b0945.